node/network/bridge/src/network.rs

```diff
@@ -235,16 +236,28 @@ impl Network for Arc<NetworkService<Block, Hash>> {
 
 		let peer_id = match peer {
 			Recipient::Peer(peer_id) =>  Some(peer_id),
-			Recipient::Authority(authority) =>
-				authority_discovery
-				.get_addresses_by_authority_id(authority)
-				.await
-				.and_then(|addrs| {
-					addrs
-						.into_iter()
-						.find_map(|addr| peer_id_from_multiaddr(&addr))
-				}),
-			};
+			Recipient::Authority(authority) => {
+				let mut found_peer_id = None;
+				// Note: `get_addresses_by_authority_id` searched in a cache, and it thus expected
+				// to be very quick.
+				for addr in authority_discovery
+					.get_addresses_by_authority_id(authority).await
+					.into_iter().flat_map(|list| list.into_iter())
+				{
+					let (peer_id, addr) = match parse_addr(addr) {
+						Ok(v) => v,
+						Err(_) => continue,
+					};
+					NetworkService::add_known_address(
```

Comment:
Two questions:
1. Why do we only need this when we want to connect to authorities?
2. Don't we keep a DHT to resolve `PeerId`s to addresses? I mean if we have them cached locally, it would be stupid to not use them, but it should still work if we did not - right?